src/meltano/core/tracking/environment.py

```diff
@@ -0,0 +1,124 @@
+"""Environment context for the Snowplow tracker."""
+
+from __future__ import annotations
+
+import ctypes
+import ctypes.wintypes
+import os
+import platform
+import uuid
+from collections import defaultdict
+from datetime import datetime
+from pathlib import Path
+from platform import system
+from typing import Any
+
+import psutil
+from backports.cached_property import cached_property
+from snowplow_tracker import SelfDescribingJson
+from structlog.stdlib import get_logger
+
+import meltano
+
+logger = get_logger(__name__)
+
+# This file is only ever created in CI when building a release
+release_marker_path = Path(__file__).parent / ".release_marker"
+
+
+class EnvironmentContext(SelfDescribingJson):
+    def __init__(self):
+        logger.debug(
+            f"Initializing '{type(self).__module__}.{type(self).__qualname__}'"
+        )
+        ci_markers = ('GITHUB_ACTIONS', 'CI')
```

Comment:
I was going to say to remove this if we don't have confidence, but actually, this looks valid, and valuable.

- https://adamj.eu/tech/2020/03/09/detect-if-your-tests-are-running-on-ci/